pkg/cmd/cli/secrets/new.go

```diff
@@ -144,18 +147,22 @@ func (o *CreateSecretOptions) Validate() error {
 		return errors.New("at least one source file or directory must be specified")
 	}
 
-nameCheck:
-	switch o.SecretTypeName {
-	case string(kapi.SecretTypeOpaque), "":
-		// this is ok
-	default:
-		// TODO this probably isn't a good idea.  It limits the power of this command.  Maybe allow unknown names with a force?
-		for _, secretType := range KnownSecretTypes {
-			if o.SecretTypeName == string(secretType.Type) {
-				break nameCheck
+	if !o.AllowUnknownTypes {
```

Comment:
> I feel like I would always want this to be true.

`oc secrets new --type=kuberentes/dockercfg blah blah`  followed by "why doesn't my secret work?"  There's a small set of enumerate strings that have server side validation for "do I know this secret".